### PR TITLE
images: Add missing bits for creating an arch package

### PR DESCRIPTION
**Pull request by jelly**
Implement the correct commands for generating an Arch Linux cockpit
package.

### CODE REVIEW

File: images/scripts/arch.install [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#! /bin/bash
+
+set -e
+
+/var/lib/testvm/arch.install "$@"
```
Comment (martinpitt) [inline images/scripts/arch.install:5]: Arch does not have multiple releases, and there is no parameterization here. Please just move arch.install from scripts/lib/ to scripts and avoid the indirection.

File: images/scripts/lib/arch.install [hunk 2]
```diff
@@ -42,9 +42,20 @@ if [ -n "$do_build" ]; then
...
+    upstream_ver=$(echo "$dist_tar" | sed 's/^.*-//; s/.tar\..z//' | head -n1)
+
+    rm -rf cockpit-*/
+    tar -xf $dist_tar
+    ( cd $resultdir/
+      cp ../cockpit-*/tools/arch/* .
+      sed -i "s/SOURCE/$dist_tar/" PKGBUILD
+      sed -i "s/VERSION/$upstream_ver/" PKGBUILD
+      # Source needs to be next to the PKGBUILD
+      cp ../$dist_tar .
```
Comment (martinpitt) [inline images/scripts/lib/arch.install:54]: Would a `mv` be enough?

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks!

**Review by martinpitt — DISMISSED** [review]
Danke! Feel free to land if you tested `test/image-prepare arch` locally, otherwise run an `arch@cockpit-project/cockpit` test here. It'll most likely fail, but it should at least get past image-prepare.

**Comment by jelly** [comment]
> Danke! Feel free to land if you tested `test/image-prepare arch` locally, otherwise run an `arch@cockpit-project/cockpit` test here. It'll most likely fail, but it should at least get past image-prepare.

Tested it locally, still needs a PR against cockpit for the missing packaging bits. I'll open a PR there and then trigger the test.

**Comment by jelly** [comment]
> > Danke! Feel free to land if you tested `test/image-prepare arch` locally, otherwise run an `arch@cockpit-project/cockpit` test here. It'll most likely fail, but it should at least get past image-prepare.
> 
> Tested it locally, still needs a PR against cockpit for the missing packaging bits. I'll open a PR there and then trigger the test.

This PR https://github.com/cockpit-project/cockpit/pull/16547 now proves the tests can run with the changes in this PR.
https://logs.cockpit-project.org/logs/pull-16547-20211102-100749-e0a2e91d-arch-bots-2582/log.html

**Review by martinpitt — APPROVED** [review]
Nice, thanks! (and ugh, so many failures)

**Pull request merged**